plugins/opsgenie/plugin.spec.yaml

```diff
@@ -0,0 +1,277 @@
+plugin_spec_version: v2
+extension: plugin
+products: [insightconnect]
+name: opsgenie
+title: OpsGenie
+description: Notify the right people, reduce response time, and avoid alert fatigue within a workflow
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+supported_versions: ["2022-01-11"]
+status: []
+resources:
+  source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/plugins/opsgenie
+  license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url: https://www.atlassian.com/pl/software/opsgenie
+tags:
+  - atlassian
+  - opsgenie
+  - alerts
+  - notifications
+hub_tags:
+  use_cases: [alerting_and_notifications]
+  keywords: [atlassian]
+  features: []
+enable_cache: false
+connection:
+  api_key:
+    title: Authorization API Key
+    description: OpsGenie authorization API key
+    type: credential_secret_key
+    required: true
+    example: "1234567e-123c-123c-123c-1234567e9xAd"
+actions:
+  create_alert:
+    title: Create Alert
+    description: Creates an alert for OpsGenie
+    input:
+      message:
+        title: Message
+        description: Message of the alert
+        type: string
+        required: true
+        example: "An example alert message"
+      alias:
+        title: Alias
+        description: Client-defined identifier of the alert, that is also the key element of Alert deduplication
+        type: string
+        required: false
+        example: "An example alias"
+      description:
+        title: Description
+        description: Description field of the alert that is generally used to provide a detailed information about the alert
+        type: string
+        required: false
+        example: "An example description"
+      responders:
+        title: Responders
+        description: 'Teams, users, escalations and schedules that the alert will be routed to send notifications. "id/name": Either id or name of each responder should be provided. "type": team, user, escalation, schedule. Format: [{"id/name":"value", "type":"team/user/escalation/schedule"}]'
+        type: "[]object"
+        required: false
+        example: '[{"id":"4513b7ea-3b91-438f-b7e4-e3e54af9147c", "type":"team"},{"name":"NOC","type":"team"}]'
+      visibleTo:
+        title: VisibleTo
+        description: Teams and users that the alert will become visible to without sending any notification. Type field is mandatory for each item, where possible values are team and user. In addition to the type field, either ID or name should be given for teams and either ID or username should be given for users. Please note that alert will be visible to the teams that are specified within responders field by default, so there is no need to re-specify them within visibleTo field. You can refer below for example values
+        type: "[]object"
+        required: false
+        example: '[{"id":"4513b7ea-3b91-438f-b7e4-e3e54af9147c","type":"team"},{"name":"example_name","type":"team"}]'
```

Comment:
This should also be split into multiple input fields. As a general rule of thumb, avoid JSON object inputs wherever possible.

Reply:
We think the same but we don't know exactly how to split a list of objects into many input variables with enum as id/name and type. Could you please tell us a little how to do that properly? Would be great with a little example code 😃